src/main/java/melchizedek/Melchizedek.java

```diff
@@ -0,0 +1,150 @@
+
+            printSeparator();
+
+            switch (tokens[0].toLowerCase()) {
+            case "bye":
+                FileHandling.writeToFile(taskList);
+                sayByeToUser();
+                return;
+
+            case "help":
+                listCommands();
+                break;
+
+            case "list":
+                taskList.printTaskList();
+                break;
+
+            case "mark":
+                try {
+                    taskList.markTaskAsDone(Integer.parseInt(tokens[1]));
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("\tOh no! Please specify which task number to mark.");
+                    System.out.println("\tExample: mark 3");
+                } catch (InvalidTaskNumberException e) {
+                    System.out.println("\tUh oh! Please input a valid task number!");
+                    taskList.printNumberOfTasks();
+                }
+                break;
+
+            case "unmark":
+                try {
+                    taskList.unmarkTaskAsDone(Integer.parseInt(tokens[1]));
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("\tOh no! Please specify which task number to unmark.");
+                    System.out.println("\tExample: unmark 2");
+                } catch (InvalidTaskNumberException e) {
+                    System.out.println("\tUh oh! Please input a valid task number!");
+                    taskList.printNumberOfTasks();
+                }
+                break;
+
+            case "todo":
+                try {
+                    if (tokens.length < 2) {
+                        throw new DescriptionNotPresentException();
+                    }
+                    taskList.addTodo(Arrays.copyOfRange(tokens, 1, tokens.length));
+                } catch (DescriptionNotPresentException e) {
+                    System.out.println("\tUh oh! I cannot create a todo with no description!");
+                    System.out.println("\tExample: todo read lecture notes");
+                }
+                break;
+
+            case "deadline":
+                try {
+                    if (tokens.length < 2) {
+                        throw new DescriptionNotPresentException();
+                    }
+                    taskList.addDeadline(Arrays.copyOfRange(tokens, 1, tokens.length));
+                } catch (DescriptionNotPresentException e) {
+                    System.out.println("\tUh oh! I cannot create a deadline with no description!");
+                    System.out.println("\tExample: deadline coding assignment /by 12pm");
+                }
+                break;
+
+            case "event":
+                try {
+                    if (tokens.length < 2) {
+                        throw new DescriptionNotPresentException();
+                    }
+                    taskList.addEvent(Arrays.copyOfRange(tokens, 1, tokens.length));
```

Comment:
Maybe it would be good to refactor this magic literal `1` that occurs in several places!